### PR TITLE
Change subsystem to "storage" in remote_read_queries gauge

### PR DESCRIPTION
**Pull request by roidelapluie**
Updated the Prometheus gauge definition for `remote_read_queries` to reflect the correct subsystem ("storage" instead of "api").

### CODE REVIEW

**Review by beorn7 — DISMISSED** [review]

File: storage/remote/read_handler.go
Comment (machine424) [inline storage/remote/read_handler.go:61]: The read client also exposed a `remote_read_queries` metric under the `remote_storage` subsystem https://github.com/prometheus/prometheus/blob/ff0a1e5e1165caa472c522f6dc25281d7e6a483f/storage/remote/client.go#L83-L90, is this ok?

Comment (machine424) [reply storage/remote/read_handler.go]: The change now yields `prometheus_remote_read_handler_remote_read_queries`, I think we can remove the second `remote`, to make it `prometheus_remote_read_handler_read_queries` or even `prometheus_remote_read_handler_queries`. (or even get back `prometheus_remote_read_storage_queries` if we make the changes below)

I think we can leverage `3.0` to also clarify the others in `client.go`

Currently, `remoteReadQueriesTotal` yields `prometheus_remote_storage_read_queries_total`
and `remoteReadQueries` translates to `prometheus_remote_storage_remote_read_queries`.

I think `storage` has nothing to do there, and it'd be clearer if we mention `client`:

- `prometheus_remote_storage_read_queries_total` -> `prometheus_remote_read_client_queries_total`
- `prometheus_remote_storage_remote_read_queries` -> `prometheus_remote_read_client_queries`

WDYT?

**Comment by beorn7** [comment]
I don't feel qualified to judge the remote read metrics naming. People actually dealing with remote read should do that. @bwplotka perhaps?

In any case, before we merge this, the commit description needs to be updated to what we are actually doing here.

File: storage/remote/read_handler.go
Comment (bwplotka) [inline storage/remote/read_handler.go:61]: why not `storage` as you said in PR title?

Comment (machine424) [reply storage/remote/read_handler.go:61]: https://github.com/prometheus/prometheus/pull/14642#discussion_r1716712415 :)

**Comment by jan--f** [comment]
We have merge `release-3.0` into `main`. The base branch for this PR should be changed to `main` now.

**Review by bwplotka — CHANGES_REQUESTED** [review]
I don't get why such a long metric name, and not just storage or something. It's literally called remote read metric, isn't that obvious it's a remote handler?

File: storage/remote/read_handler.go
Comment (bwplotka) [reply storage/remote/read_handler.go]: I like the client approach, LGTM for that!

**Comment by krajorama** [comment]
Hello from the bug scrub. @bwplotka please consider reviewing / approving again, this seems to be blocked on you having requested changes. Or in fact consider closing that as it missed the window for breaking changes in 3.0.

**Comment by beorn7** [comment]
Technically, we can change metrics without bumping the major version number. https://prometheus.io/docs/prometheus/latest/stability/#api-stability-guarantees says: "The metrics in the /metrics endpoint of Prometheus itself" are considered unstable.

**Comment by machine424** [comment]
I opened https://github.com/prometheus/prometheus/pull/15541

**Comment by beorn7** [comment]
Closing this as #15541 supersedes it.

**Pull request closed without merging**
